ouroboros-consensus/test-storage/Test/Ouroboros/Storage/ImmutableDB/StateMachine.hs

```diff
@@ -233,6 +236,15 @@ run ImmutableDBEnv { db, internal, registry, varNextId } runCorruption its cmd =
       closeDB db
       reopen db valPol
       Tip <$> getTip db
+    ReopenInThePast valPol curSlot -> do
+      mapM_ iteratorClose (unWithEq <$> its)
+      closeDB db
+      atomically $ writeTVar varCurSlot curSlot
```

Comment:
Comment (somewhere): derived from the tip of the ImmutableDB (model) not from the blockchain time.